### PR TITLE
5320 devops ci cd

### PR DESCRIPTION
**Pull request by digorgonzola**

### CODE REVIEW

**Comment by digorgonzola** [comment]
@vietnguyengit I've had to change your multi-stage docker build approach in this one in order to facilitate publishing the java package to code artifact.

File: .gitignore [hunk 1]
```diff
@@ -33,6 +33,4 @@ build/
 ### VS Code ###
 .vscode/
 
-.mvn
```
Comment (vietnguyengit) [inline .gitignore:36]: are you taking this off on purpose?

File: .github/environment/staging.env [hunk 2]
```diff
@@ -0,0 +1,5 @@
+AWS_REGION=ap-southeast-2
+CODEARTIFACT_DOMAIN=gamma-aodn-org-au
+CODEARTIFACT_REPO=maven-aodn-store
+ECR_REGISTRY=450356697252.dkr.ecr.ap-southeast-2.amazonaws.com
+ECR_REPOSITORY=es-indexer
```
Comment (vietnguyengit) [inline .github/environment/staging.env:5]: I'm just curious why you don't declare these environments inside the settings tab of the repo but declaring them in files instead?

File: Dockerfile [hunk 3]
```diff
@@ -1,21 +1,8 @@
...
-
-#########################3
-
-# Stage 2: Run the Spring Boot application
 FROM amazoncorretto:17
 
 WORKDIR /app
 
-COPY --from=builder /app/target/es-indexer-0.0.1-SNAPSHOT.jar /app/app.jar
+COPY ./target/es-indexer-*.jar /app/app.jar
```
Comment (vietnguyengit) [inline Dockerfile:5]: cool

**Review by vietnguyengit — COMMENTED** [review]
cow

File: .github/environment/staging.env (hunk 2 above)
Comment (digorgonzola) [reply .github/environment/staging.env:5]: Ah these are just here as a record. It's easy then to see what the values should be and you can also update the actual environment variables using the GH CLI (as per the README)

File: .gitignore (hunk 1 above)
Comment (digorgonzola) [reply .gitignore:36]: Yes so we can enable the maven git versioning extension

**Review by vietnguyengit — APPROVED** [review]
LGTM

**Pull request merged**